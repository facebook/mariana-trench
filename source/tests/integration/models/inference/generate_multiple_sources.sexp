(method (private) "LFlow;.flow:()LData;"
 (
  LINE(1)
  (const v0 1)

  LINE(2)
  (if-eqz v0 :else)
  (invoke-virtual (v1) "LSource;.source:()LData;")
  (move-result-object v0)

  LINE(3)
  (return-object v0)
  (goto :out)

  LINE(4)
  (:else)
  (invoke-virtual (v1) "LSource;.source:()LData;")
  (move-result-object v0)

  LINE(5)
  (:out)
  (return-object v0)
 )
)
