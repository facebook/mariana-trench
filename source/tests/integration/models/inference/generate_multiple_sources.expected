{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "generations" :
      [
        {
          "call_position" :
          {
            "line" : 2,
            "path" : "Flow.java"
          },
          "callee" : "LSource;.source:()LData;",
          "callee_port" : "Return",
          "caller_port" : "Return",
          "distance" : 1,
          "kind" : "TestSource",
          "origins" :
          [
            "LSource;.source:()LData;"
          ]
        },
        {
          "call_position" :
          {
            "line" : 4,
            "path" : "Flow.java"
          },
          "callee" : "LSource;.source:()LData;",
          "callee_port" : "Return",
          "caller_port" : "Return",
          "distance" : 1,
          "kind" : "TestSource",
          "origins" :
          [
            "LSource;.source:()LData;"
          ]
        }
      ],
      "method" : "LFlow;.flow:()LData;",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    }
  ]
}
